### PR TITLE
Initialize column range map with mutable list

### PR DESCRIPTION
**Pull request by ellisjoe**
**Goals (and why)**:

**Implementation Description (bullets)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3147)

### CODE REVIEW

**Comment by sandorw** [comment]
Filed https://github.com/palantir/atlasdb/issues/3148 as a follow-up to add test coverage

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java [hunk 1]
```diff
@@ -627,7 +627,7 @@ private void verifyColumnRanges(Transaction readOnlyTransaction) {
                     if (rangesToRows.get(range) != null) {
                         rangesToRows.get(range).add(row);
                     } else {
-                        rangesToRows.put(range, ImmutableList.of(row));
+                        rangesToRows.put(range, Lists.newArrayList(row));
```
Comment (schlosna) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java]: could simplify the whole if/else to:

```
                    rangesToRows.computeIfAbsent(range, ignored -> Lists.newArrayList()).add(row);
```

Comment (ellisjoe) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java]: I'm happy either way. @sandorw?

Comment (sandorw) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java]: Going to keep this for now for the sake of merging and getting the release out. I'll bundle it with another PR which is fixing some other code style in the same class (https://github.com/palantir/atlasdb/pull/3149).

**Pull request merged**